pkg/proxy/iptables/proxier.go

```diff
@@ -835,15 +836,24 @@ func (proxier *Proxier) syncProxyRules() {
 	serviceUpdateResult := proxier.serviceMap.Update(proxier.serviceChanges)
 	endpointUpdateResult := proxier.endpointsMap.Update(proxier.endpointsChanges)
 
+	// We need to detect stale Services for UDP so we can clean
+	// dangling conntrack entries that can blackhole traffic.
 	staleServices := serviceUpdateResult.UDPStaleClusterIP
+	staleNodePorts := sets.NewInt()
 	// merge stale services gathered from updateEndpointsMap
+	// an UDP service that changes from 0 to non-0 endpoints is considered stale.
 	for _, svcPortName := range endpointUpdateResult.StaleServiceNames {
```

Comment:
OK... the reason this needs so many comments is that the (pre-existing) naming is terrible.

`serviceUpdateResult.UDPStaleClusterIP` is ok; that's "UDP Services that used to exist but don't any more, so any references to those Cluster IPs are stale". And assigning that to a variable called `staleServices` therefore makes sense.

But `endpointUpdateResult.StaleServiceNames` is "UDP Services that used to have 0 endpoints but now have more than 0 endpoints". `StaleServiceNames` doesn't explain that at all; the connection to staleness is that there might be stale conntrack entries for the service, not that the service itself is in any way stale... So adding those services to `staleServices` is confusing... (although it's correct, because `staleServices` actually just means "services whose conntrack entries we will delete later").

And then `staleNodePorts` means "newly-added UDP node ports" and is even more confusing...

So... this would be much clearer if you renamed `staleServices` to `staleConntrackServices` or `conntrackCleanupServices` or something and likewise with `staleNodePorts`. The use of "stale" in the log messages below could be made clearer as well.

And `StaleServiceNames` ought to be `NewlyActiveUDPServices` or something like that but maybe not in this PR

Reply:
done